Storage/StorageTests/CoreData/CoreDataIterativeMigratorTests.swift

```diff
@@ -260,6 +261,34 @@ final class CoreDataIterativeMigratorTests: XCTestCase {
                            storeFolderURL.appendingPathComponent(fileName).path)
         }
     }
+
+    func test_iterativeMigrate_will_not_migrate_if_the_database_and_the_model_are_compatible() throws {
+        // Given
+        let model = try managedObjectModel(for: "Model 28")
+
+        // Start a container to create an existing database file.
+        let storeURL = try urlForStore(withName: "Woo_Compatibility_Test.sqlite", deleteIfExists: true)
+        let _ = try startPersistentContainer(storeURL: storeURL, storeType: NSSQLiteStoreType, model: model)
+
+        let spyFileManager = SpyFileManager()
+        let migrator = CoreDataIterativeMigrator(modelsInventory: modelsInventory, fileManager: spyFileManager)
+
+        // When
+        // Migrate up to the same model version.
+        let (result, debugMessages) = try migrator.iterativeMigrate(sourceStore: storeURL,
+                                                                    storeType: NSSQLiteStoreType,
+                                                                    to: model)
+
+        // Then
+        XCTAssertTrue(result)
+
+        XCTAssertEqual(debugMessages.count, 1)
+        assertThat(try XCTUnwrap(debugMessages.first), contains: "No migration necessary.")
+
+        // There should be no file operations.
+        assertEmpty(spyFileManager.deletedItems)
+        XCTAssertEqual(spyFileManager.movedItems.count, 0)
```

Comment:
Mainly because `assertEmpty` accepts an `Array` and `movedItems` is a Dictionary. So I gave up right away. 😄 

But thanks for the nudge, I changed `assertEmpty` so it accepts a `Collection`. 93ea48502c40f176c6cc40e791db340dd8d718c1 and modified this line to use `assertEmpty()` ccb007dd078012a9a4e5ef65735c36939e989f32.

Reply:
ooh I forgot that these two are different types! awesome update 💯